src/internal.c

```diff
@@ -3550,6 +3612,11 @@ int InitSSL(WOLFSSL* ssl, WOLFSSL_CTX* ctx)
         return MEMORY_E;
     }
     XMEMSET(ssl->arrays, 0, sizeof(Arrays));
+    ssl->arrays->preMasterSecret = (byte*)XMALLOC(ENCRYPT_LEN, ssl->heap,
+        DYNAMIC_TYPE_TMP_BUFFER);
+    if (ssl->arrays->preMasterSecret == NULL) {
+        return MEMORY_E;
+    }
```

Comment:
Updated to zero PMS. Its dynamic so the QAT hardware can use the buffer directly.